wpilibc/athena/include/vision/VisionRunner.h

```diff
@@ -0,0 +1,41 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) FIRST 2016. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#pragma once
+
+#include <functional>
+
+#include "ErrorBase.h"
+#include "cscore.h"
+#include "opencv2/core/mat.hpp"
+#include "vision/VisionPipeline.h"
+
+namespace frc {
+
+template <typename T>
```

Comment:
I couldn't find a way to do it without assertions. It will fail to compile if the object does not contain a Process function. But it doesn't specifically have to implement that interface. It just has to have the method.